core/controllers/topics_and_skills_dashboard.py

```diff
@@ -59,12 +59,12 @@ def get(self):
         skill_summary_dicts = [
             summary.to_dict() for summary in skill_summaries]
 
-        topic_rights_list = topic_services.get_all_topic_rights()
-        for topic_rights in topic_rights_list:
-            for topic_summary in topic_summary_dicts:
-                if topic_summary['id'] == topic_rights.id:
-                    topic_summary[
-                        'is_published'] = topic_rights.topic_is_published
+        topic_rights_dict = topic_services.get_all_topic_rights()
+        for topic_summary in topic_summary_dicts:
+            topic_rights = topic_rights_dict[topic_summary['id']]
+            if topic_rights:
+                topic_summary[
```

Comment:
For the future: you can also do

    topic_summary['is_published'] = (
        topic_rights.topic_is_published)

That said ... shouldn't this be topic_summary['topic_is_published'], similar to above?